nowcasting_dataset/dataset/split/split.py

```
@@ -0,0 +1,80 @@
+    method: SplitMethod,
+    train_test_validation_split: Tuple[int] = (3, 1, 1),
+    train_test_validation_year: TrainValidationTestYear = default_train_test_validation_year,
+) -> (List[pd.Timestamp], List[pd.Timestamp], List[pd.Timestamp]):
+    """
+    Split the date using various different methods
+
+    Args:
+        datetimes: The datetimes to be split
+        method: the method to be used
+        train_test_validation_split: ratios of how the split is made
+        train_test_validation_year: pydantic class of which years below to which dataset
+
+    Returns: train, validation and test dataset
+
+    """
+
+    logger.info(f"Splitting data with method {method}")
+
+    if method == SplitMethod.Same:
+        train_datetimes = datetimes
+        validation_datetimes = datetimes
+        test_datetimes = datetimes
+    elif method == SplitMethod.Day:
+        train_datetimes, validation_datetimes, test_datetimes = split_day(
+            datetimes=datetimes, train_test_validation_split=train_test_validation_split
+        )
+    elif method == SplitMethod.DayRandom:
+        train_datetimes, validation_datetimes, test_datetimes = split_day_random(
+            datetimes=datetimes, train_test_validation_split=train_test_validation_split
+        )
+    elif method == SplitMethod.Week:
+        train_datetimes, validation_datetimes, test_datetimes = split_week(
+            datetimes=datetimes, train_test_validation_split=train_test_validation_split
+        )
+    elif method == SplitMethod.WeekRandom:
+        train_datetimes, validation_datetimes, test_datetimes = split_week_random(
+            datetimes=datetimes, train_test_validation_split=train_test_validation_split
+        )
+    elif method == SplitMethod.Year:
+        train_datetimes, validation_datetimes, test_datetimes = split_year(
+            datetimes=datetimes, train_test_validation_year=train_test_validation_year
+        )
+    else:
+        raise
```

If you `raise` outside of a `try... except...` block then Python will complain `RuntimeError: No active exception to reraise` :)

Maybe replace `raise` with something like `raise ValueError('Unknown SplitMethod!')`?

---

yea good point